### PR TITLE
Add flow interface

### PR DESCRIPTION
**Pull request by lukanus**

### CODE REVIEW

File: worker/flow/handler/managerstream/connectivity/structs/pool.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package structs
+
+// TaskResponseChanPool is default pool of channels TaskResponse
+var TaskResponseChanPool = NewtaskResponsePool(40)
```
Comment (deelawn) [inline worker/flow/handler/managerstream/connectivity/structs/pool.go:4]: Capitalize the "t" in `Newtask`

File: worker/flow/handler/managerstream/connectivity/structs/pool.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+type taskResponsePool struct {
+	pool chan chan TaskResponse
+}
+
+// NewtaskResponsePool is taskResponsePool constructor
+func NewtaskResponsePool(size int) *taskResponsePool {
```
Comment (deelawn) [inline worker/flow/handler/managerstream/connectivity/structs/pool.go:11]: Capitalize the "t" in `Newtask`

File: worker/flow/handler/managerstream/connectivity/structs/structs.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	defer sa.respLock.RUnlock()
+	defer sa.reqLock.Unlock()
+
+	if sa.State == StreamOffline {
+		return nil
+	}
+	close(sa.Finish)
+
+	TaskResponseChanPool.Put(sa.ResponseListener)
+	close(sa.RequestListener)
```
Comment (deelawn) [inline worker/flow/handler/managerstream/connectivity/structs/structs.go:91]: Should `sa.State` be set to `StreamOffline`?

File: worker/flow/handler/managerstream/connectivity/structs/structs.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	Id      uuid.UUID
+	Type    string
+	Order   uint64
+	Final   bool
+	Error   TaskError
+	Payload json.RawMessage
+}
+
+// IndexerClienter is a client interface
+type IndexerClienter interface {
```
Comment (deelawn) [inline worker/flow/handler/managerstream/connectivity/structs/structs.go:127]: What is a `Clienter`? I know that go likes to have interfaces named with "-er"-like suffixes.  If we want to do that then something like `IndexerClientStreamRegistrar` makes more sense.  I'm also fine with keeping it how it is; my proposed name is kind of long.

File: worker/process/rangedv2/rangedv2.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
+package rangedv2
```
Comment (deelawn) [inline worker/process/rangedv2/rangedv2.go:1]: Out of curiosity, what has changed from ranged v1?

File: worker/flow/handler/managerstream/transport/grpc/indexer/indexer.pb.go [hunk 6]
```diff
@@ -0,0 +1,499 @@
+// Code generated by protoc-gen-go. DO NOT EDIT.
```
Comment (deelawn) [inline worker/flow/handler/managerstream/transport/grpc/indexer/indexer.pb.go:1]: Where is the proto file used to generate this?

**Review by deelawn — APPROVED** [review]
Looks good to me.

I think that most of the comments I made are for existing code that was just moved here from the manager, but it was my first time looking at the code and there are some bugs and typos that can maybe be fixed.

File: worker/flow/handler/managerstream/connectivity/connectivity.go [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+	for {
+		select {
+		case <-ctx.Done():
+			tckr.Stop()
+			return
+		case <-tckr.C:
+			wc.managerAddressesLock.RLock()
+			for _, ad := range wc.managerAddresses {
+				readr.Seek(0, 0)
```
[the remark this answers is not in the document]
Comment (lukanus) [reply worker/flow/handler/managerstream/connectivity/connectivity.go:109]: Saddly io.ReadSeeker is not widely available  ;)

File: worker/flow/handler/managerstream/connectivity/connectivity.go [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+				timer := metrics.NewTimer(ad.Observer)
+				req, err := http.NewRequestWithContext(ctx, http.MethodPost, "http://"+ad.Address, readr)
+				if err != nil {
+					workerStatus.WithLabels(ad.Address, "600", "err_creating_request").Inc()
```
[the remark this answers is not in the document]
Comment (lukanus) [reply worker/flow/handler/managerstream/connectivity/connectivity.go:113]: We had a dashboard for this one long time ago, I think when we stabilize "everything" we should put new ones anyway :)

File: worker/flow/handler/managerstream/managerstream.go [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+// RegisterStream adds new listeners to the streams - currently fixed number per stream
+func (ic *IndexerClient) RegisterStream(ctx context.Context, stream *cStructs.StreamAccess) error {
+	ic.logger.Debug("Register Stream", zap.Stringer("streamID", stream.StreamID))
+
+	ic.sLock.Lock()
+	defer ic.sLock.Unlock()
+	ic.streams[stream.StreamID] = stream
+
+	// Limit workers not to create new goroutines over and over again
+	for i := 0; i < 20; i++ {
```
[the remark this answers is not in the document]
Comment (lukanus) [reply worker/flow/handler/managerstream/managerstream.go:66]: Well it wasn't for a long time :) I think this code will change anyway on next iteration

File: worker/flow/handler/managerstream/managerstream.go [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+		})
+		return
+	}
+
+	ic.logger.Debug("Task request is subtyped", zap.String("subtype", heightRange.SubKind))
+	kind, payload, err := ic.f.GetData(ctx, heightRange)
+	if err != nil {
+		stream.Send(cStructs.TaskResponse{
+			Id:    taskRequest.Id,
+			Error: cStructs.TaskError{Msg: fmt.Sprintf("error processing  %s ", heightRange.SubKind, taskRequest.Type)},
```
[the remark this answers is not in the document]
Comment (lukanus) [reply worker/flow/handler/managerstream/managerstream.go:121]: :+1

File: worker/flow/handler/managerstream/connectivity/structs/structs.go (hunk 4 above)
Comment (lukanus) [reply worker/flow/handler/managerstream/connectivity/structs/structs.go:127]: Well, let's revisit this one on next iteration :) It's pure copy paste right now :)

File: worker/flow/handler/managerstream/connectivity/structs/structs.go (hunk 3 above)
Comment (lukanus) [reply worker/flow/handler/managerstream/connectivity/structs/structs.go:91]: Let's just not change it right now. This whole thing would e probably not needed :)

File: worker/flow/handler/managerstream/connectivity/metrics.go [hunk 11]
```diff
@@ -0,0 +1,21 @@
+package connectivity
+
+import "github.com/figment-networks/indexing-engine/metrics"
+
+var (
```
[the remark this answers is not in the document]
Comment (lukanus) [reply worker/flow/handler/managerstream/connectivity/metrics.go:5]: This was used really early on :) and mostly when manager was receiving real data, not just confirmations

File: worker/flow/handler/managerstream/managerstream.go [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+			ic.sLock.Unlock()
+			return
+		case <-stream.Finish:
+			return
+		case taskRequest := <-stream.RequestListener:
+			nCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+			switch taskRequest.Type {
+			case structs.ReqIDGetLatestMark:
+				ic.GetLatestMark(nCtx, taskRequest, stream)
+			case structs.ReqIDLatestData:
```
[the remark this answers is not in the document]
Comment (lukanus) [reply worker/flow/handler/managerstream/managerstream.go:89]: Good idea!

File: worker/process/rangedv2/rangedv2.go (hunk 5 above)
Comment (lukanus) [reply worker/process/rangedv2/rangedv2.go:1]: I removed great deal of complexity there. 
It's just a function right now, previously you were assigning structure to ranger interface. But with the latest changes in the flow it's not really needed anymore

**Pull request merged**